### DISCUSSION TITLE
Is there any built in way to clone entire graph

### DISCUSSION THREAD

**Discussion by Sanjeevraghav** [antvis/X6#1827] [Q&A]
We need to clone the entire graph but node and edge ids must be unique. Please let me know is there any built-in feature in the library?

**Comment by niemes** [comment]
Hello,

there is a method to get the graph as a json. Depends of whats in your graph. This method will return the cell and edges. But if your graph have some custom nodes you will need to grab them using the data available with this method. 

```javascript
let currentData = graph.toJSON(...)
console.log(graph.parseJSON(currentData))
```

**source**: [x6 Doc](https://x6.antv.vision/en/docs/api/graph/model#tojson)

Hope it helps o/
